### PR TITLE
Update dev-machine-setup.md

### PR DESCRIPTION
**Pull request by isfukuda**
## Reason for This PR

`[Author TODO: add issue # or explain reasoning.]`
Improve security for VM on GCP/GCE, and add VM on Azure Case.
## Description of Changes

`[Author TODO: add description of changes.]`
Some security tips add. Create new firewall rules and attach it to VM (GCP/GCE).
On Azure, using "Nested support Instance Type"  with Azure CLI. 

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any API changes are reflected in `firecracker/swagger.yaml`.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

**Review by serban300 — COMMENTED** [review]
This doc seems a bit inconsistent. For AWS we provide setup steps using the graphical interface. For Google cloud we provide steps using de CLI. Now for Azure we use the CLI again. I wonder, wouldn't it be better to just drop a link to the official Azure documentation for creating and setting up a VM ? And just mention a few details, for example the fact that the instance needs to support nested virtualization ? I suppose the official documentation is much more detailed than what we can provide. Also the official doc should contain steps both for the graphical interface and the CLI.

**Comment by isfukuda** [comment]
@serban300 
Thanks for review and the comment. 
>  I wonder, wouldn't it be better to just drop a link to the official Azure documentation for creating and setting up a VM ?
OK I'll do.

**Comment by dianpopa** [comment]
hello @isfukuda are you still interested in merging the PR? Thanks!

**Comment by isfukuda** [comment]
Hello @dianpopa !
Sorry for late for updates. I need a few days more. Thank you for your corporation.

File: docs/dev-machine-setup.md [hunk 1]
```diff
@@ -216,4 +237,88 @@ In a nutshell, setting up a GCP account involves the following steps:
...
+Another options to set up Firecracker for development purposes is to use
+a VM on Microsoft Azure (VirtulMachine), which supports nested virtualization
+and allows to run KVM. If you don't have a Microsoft (Azure) account, you can 
+find brief instructions in [document](https://azure.microsoft.com/en-us/get-started/).
+
+Here is a brief summary of steps to create such a setup (full instructions to
+set up a Ubuntu-based VM on GCE with nested KVM enablement can be found in Azure
+[documentation](https://azure.microsoft.com/en-us/blog/nested-virtualization-in-azure/)
+
+  1. Sign in for Azure CLI
```
Comment (dianpopa) [inline docs/dev-machine-setup.md:249]: A first step here would be:

1. Install Azure CLI
   More details on how to do that [here](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)

File: docs/dev-machine-setup.md [hunk 2]
```diff
@@ -216,4 +237,88 @@ In a nutshell, setting up a GCP account involves the following steps:
...
+a VM on Microsoft Azure (VirtulMachine), which supports nested virtualization
+and allows to run KVM. If you don't have a Microsoft (Azure) account, you can 
+find brief instructions in [document](https://azure.microsoft.com/en-us/get-started/).
+
+Here is a brief summary of steps to create such a setup (full instructions to
+set up a Ubuntu-based VM on GCE with nested KVM enablement can be found in Azure
+[documentation](https://azure.microsoft.com/en-us/blog/nested-virtualization-in-azure/)
+
+  1. Sign in for Azure CLI 
+     fllow the commands bellow , Open the LINK and paste the code, get information.
```
Comment (dianpopa) [inline docs/dev-machine-setup.md:250]: This phrase needs some rephrasing, I cannot understand what it is meant to say.

File: docs/dev-machine-setup.md [hunk 3]
```diff
@@ -216,4 +237,88 @@ In a nutshell, setting up a GCP account involves the following steps:
...
+     ```
+     $ az login 
+     ```
+  1. SET a Azure/VirtulMachines options , Notice: nested support VM SIZE!
```
Comment (dianpopa) [inline docs/dev-machine-setup.md:254]: Do you mean "VirtualMachines"?

Comment (dianpopa) [reply docs/dev-machine-setup.md:254]: What does "Notice: nested support VM SIZE!" mean?

Comment (isfukuda) [reply docs/dev-machine-setup.md]: Hi @dianpopa 
> What does "Notice: nested support VM SIZE!" mean?

firecracker is necessary "/dev/kvm". so update comments where you pointed out. Thank you.

**Comment by acatangiu** [comment]
Hi @isfukuda are you still interested in driving this PR?

File: docs/dev-machine-setup.md [hunk 4]
```diff
@@ -183,6 +183,27 @@ set up a Ubuntu-based VM on GCE with nested KVM enablement can be found in GCE
 
   Now you can continue with the Firecracker [Getting Started](getting-started.md)
   instructions to install and configure Firecracker in the new VM.
+  
+  And also, to secure your VM. SetUp new Firewal rules and attach it as soon as possible.
```
Comment (berciuliviu) [inline docs/dev-machine-setup.md:187]: I believe this could be added in a subsection under GCP, called "GCP Security"/"Securing GCP" or something similar. I am suggesting this since lines 184-185 are ending the above steps by pointing out to `getting-started.md` and security aspects could get their own space . Also, I would rephrase like 187 as "Secure your VM by setting up new Firewall rules:" and insert the steps below.

File: docs/dev-machine-setup.md [hunk 5]
```diff
@@ -237,4 +237,102 @@ In a nutshell, setting up a GCP account involves the following steps:
...
+     ```
+     $ az network nic create --resource-group $RG \
+       --name fc_nic1  \
+       --vnet-name $VNET \
+       --subnet $SUBNET_1 \
+       --network-security-group $NSG \
+       --public-ip-address fc_public
+     ```
+
+  1. Now we create the VM:
```
Comment (berciuliviu) [inline docs/dev-machine-setup.md:308]: "Now me" and "At last," from line 335 could be removed, to keep the imperative style from the above steps ( "Create", "Get", "Set ).

**Comment by dianpopa** [comment]
@isfukuda I am noticing some old feedback that is waiting your input. Are you still interested in working on this?

**Comment by dianpopa** [comment]
@isfukuda I am closing this as no activity was noticed here for a long time. Feel free to re-open it if you feel like resuming work on this. Thanks!
